Pcap++/src/MBufRawPacket.cpp

```diff
@@ -333,9 +333,9 @@ namespace pcpp
 			return;
 		}
 
-		m_MBuf = mBuf;
 		RawPacket::setRawData(rte_pktmbuf_mtod(mBuf, const uint8_t*), rte_pktmbuf_pkt_len(mBuf), timestamp,
 		                      LINKTYPE_ETHERNET);
+		m_MBuf = mBuf;
```

Comment:
This was a bug that was introduced somewhere, not sure where: when calling `RawPacket::setRawData()` it calls `MBufRawPacket::clear()` which sets `m_MBuf` to `nullptr`, so we have to set the mbuf after calling `RawPacket::setRawData()`, otherwise we get a segmentation fault

Reply:
this is a good explanation that should go in the commit log of "Fix a bug in MBufRawPacket::setMBuf"

Reply:
We run "Squash and merge" so these commits go away after merging. That's why it's better to put them as comments, so anyone can go to the PR that made the change and look